doc/cephfs/fs-volumes.rst

```diff
@@ -1340,5 +1340,28 @@ set with this id was present in the database
 
   $ ceph fs quiesce fs1 sub1 sub2 sub3 --set-id="external-id" --if-version=0
 
+
+.. _disabling-volumes-plugin:
+
+Disabling Volumes Plugin
+------------------------
+By default, on a Ceph cluster, volumes plugin is enabled as well as set as
+``always on`` module. However, in certain cases it might be necessary/favourable
+to disable it. For example, when a CephFS is in a degraded state, the volumes
+plugin commands may get accumulated in MGR instead of getting served. Which
+eventually causes policy throttles to kick in and MGR becomes unresponsive.
+
+For such cases, volumes plugin can be disabled even though it is an
+``always on`` module in MGR. To do so, run ``ceph mgr module disable volumes
+--yes-i-really-mean-it``. Do note that this command will disable operations
+and remove commands of volumes plugin since it will disable all CephFS
+services on the Ceph cluster accessed through this plugin.
+
+Before resorting to a measure as drastic as this, it is a good idea to try less
+drastic measures and then assess if the file system experience has improved due
+to it. One example of such less drastic measure is to disable asynchronous
+threads launched by volumes plugins for cloning and purging trash.
```

Comment:
Just a note here, once the changes to disable async threads is merged, please link the documentation from that from here.

Reply:
Okay, added this as a TODO on that PR.

Reply:
Marking this as resolved for now.

Reply:
s/due to it//
An example of a less-drastic measure is to disable the asynchronous threads used by the ``volumes`` plugin for purging trash and cloning.

Reply:
https://github.com/ceph/ceph/pull/60493 - Squid backport
https://github.com/ceph/ceph/pull/60494 - Reef backport
https://github.com/ceph/ceph/pull/60495 - Quincy backport